### PR TITLE
Use new way of handling clima parameters

### PR DESCRIPTION
**Pull request by charleskawczynski**
This PR upgrades Insolation.jl to use the new way of handling clima parameters, by:
 - Defining a `InsolationParameters` struct that has its own parameters, and populating them via a toml dictionary from CLIMAParameters. As a result, Insolation.jl (like all other packages) no longer needs to depend on CLIMAParameters. Our test, examples, and docs directory may still need to depend on CLIMAParameters in order to test the package with values from CLIMAParameters.

This PR has a lot of changes, but it does seem to pass all the tests. I've applied the same sort of pattern in this package as others in order to make the organization familiar to users if they dig into other packages with their own parameters.

Because there are so many changes (this has been CLIMAParameter changes in only one of many repos), it would be helpful if people could pitch in with updating the docs regarding the new handling of parameters. Maybe we can follow up with a docs fixup in a subsequent PR since all of these changes are breaking (while doc changes are not), and they need to be propagated in a quick sweep through many packages.

I've also made changes to GHA to use the latest version of julia and embraced the `import FooBar as FB` syntax, since it really helps managing scope compactly

### CODE REVIEW

File: docs/src/InsolationExamples.md [hunk 1]
```diff
@@ -24,16 +24,19 @@ diurnal_cycle(lat, lon, date, timezone, "Finland_June.png")
...
 ```@example
-using CLIMAParameters
-using CLIMAParameters.Planet
-struct EarthParameterSet <: AbstractEarthParameterSet end
-const param_set = EarthParameterSet()
+import CLIMAParameters as CP
+import Insolation
+import Insolation.Parameters as IP
+
+FT = Float64
```
Comment (claresinger) [inline docs/src/InsolationExamples.md:31]: The package should now work also with Float32. If that is what ClimaAtmos will use it might be best to use that in the docs too.

File: docs/src/InsolationExamples.md [hunk 2]
```diff
@@ -43,19 +46,24 @@ plot_day_lat_insolation(days, lats, F0, "YlOrRd", title, "insol_example1.png")
...
 ```@example
-using CLIMAParameters # hide
-using CLIMAParameters.Planet # hide
-struct EarthParameterSet <: AbstractEarthParameterSet end # hide
-const param_set = EarthParameterSet() # hide
+import CLIMAParameters as CP
+import Insolation
+import Insolation.Parameters as IP
+
+FT = Float64
```
Comment (claresinger) [inline docs/src/InsolationExamples.md:53]: Same here

File: docs/src/InsolationExamples.md [hunk 3]
```diff
@@ -43,19 +46,24 @@ plot_day_lat_insolation(days, lats, F0, "YlOrRd", title, "insol_example1.png")
...
-e0 = eccentricity_epoch(param_set) # hide
+γ0 = IP.obliq_epoch(param_set) # hide
+ϖ0 = IP.lon_perihelion_epoch(param_set) # hide
+e0 = IP.eccentricity_epoch(param_set) # hide
 days, lats, F0 = calc_day_lat_insolation(365, 180, param_set) # hide
 
 # decrease γ to 20.0°
-CLIMAParameters.Planet.obliq_epoch(::EarthParameterSet) = deg2rad(20.0)
-γ1 = obliq_epoch(param_set)
+# TODO: verify this
```
Comment (claresinger) [inline docs/src/InsolationExamples.md:64]: The plots in the docs look fine. Is that what you mean by verify here Charlie?

File: docs/src/InsolationExamples.md [hunk 4]
```diff
@@ -68,19 +76,23 @@ plot_day_lat_insolation(days, lats, F2-F0, "PRGn", title, "insol_example2b.png")
...
 ```@example
-using CLIMAParameters # hide
-using CLIMAParameters.Planet # hide
-struct EarthParameterSet <: AbstractEarthParameterSet end # hide
-const param_set = EarthParameterSet() # hide
+import CLIMAParameters as CP
+import Insolation
+import Insolation.Parameters as IP
+
+FT = Float64
```
Comment (claresinger) [inline docs/src/InsolationExamples.md:83]: Same here

File: docs/src/find_equinox_perihelion_dates.jl [hunk 5]
```diff
@@ -2,10 +2,11 @@ using Insolation
 using Plots
 using Dates
 
-using CLIMAParameters
-using CLIMAParameters.Planet
-struct EarthParameterSet <: AbstractEarthParameterSet end
-const param_set = EarthParameterSet()
+import CLIMAParameters as CP
+
+FT = Float64
```
Comment (claresinger) [inline docs/src/find_equinox_perihelion_dates.jl:7]: Same here

File: docs/src/plot_diurnal_cycle.jl [hunk 6]
```diff
@@ -4,13 +4,11 @@ using Dates
...
 
-using CLIMAParameters
-using CLIMAParameters.Planet
-struct EarthParameterSet <: AbstractEarthParameterSet end
-const param_set = EarthParameterSet()
+import CLIMAParameters as CP
 
-using CLIMAParameters: AbstractParameterSet
-const APS = AbstractParameterSet
+FT = Float64
```
Comment (claresinger) [inline docs/src/plot_diurnal_cycle.jl:9]: Same here

File: docs/src/InsolationExamples.md [hunk 7]
```diff
@@ -24,16 +24,19 @@ diurnal_cycle(lat, lon, date, timezone, "Finland_June.png")
 ## Daily-mean insolation
 ### Insolation in J2000
 ```@example
-using CLIMAParameters
-using CLIMAParameters.Planet
-struct EarthParameterSet <: AbstractEarthParameterSet end
-const param_set = EarthParameterSet()
+import CLIMAParameters as CP
```
Comment (claresinger) [inline docs/src/InsolationExamples.md:27]: Is CP used in the docs? Or why is this import here? (Sorry if I missed it - hard to see on my phone!)

File: src/Insolation.jl [hunk 8]
```diff
@@ -1,10 +1,10 @@
...
 
 using Dates, DelimitedFiles, Interpolations
-using CLIMAParameters
-using CLIMAParameters.Planet
-using CLIMAParameters: AbstractParameterSet
-const APS = AbstractParameterSet
+
+include("Parameters.jl")
+import .Parameters as IP
+const AIP = IP.AbstractInsolationParams
```
Comment (claresinger) [inline src/Insolation.jl:7]: Is `AIP` used? I saw in other places you’re using `IP.APS`, which I guess is the same thing. I would say it’s cleaner to just use `AIP`, but either way is fine, and we should delete the other one.

File: src/Parameters.jl [hunk 9]
```diff
@@ -0,0 +1,23 @@
+module Parameters
+
+abstract type AbstractInsolationParams end
+const APS = AbstractInsolationParams
+
+Base.@kwdef struct InsolationParameters{FT} <: AbstractInsolationParams
+    obliq_epoch::FT
```
Comment (claresinger) [inline src/Parameters.jl:7]: This is just being picky, but can we reorder this list? Just to have some of the more generic planet parameters all first and then all the orbital parameters which are time dependent and depend on the epoch after.

I would have it go: `year_anom`, `day`, `orbit_semimaj`, `tot_solar_irrad`, `epoch`, `mean_anom_epoch`, `eccentricity_epoch`, `obliq_epoch`, `lon_perihelion_epoch`.

File: src/Parameters.jl [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+    lon_perihelion_epoch::FT
+    day::FT
+    orbit_semimaj::FT
+    eccentricity_epoch::FT
+    tot_solar_irrad::FT
+    year_anom::FT
+    mean_anom_epoch::FT
+    epoch::FT
```
Comment (claresinger) [inline src/Parameters.jl:15]: What I would like to do is have this be a String. I’m not currently using the `epoch` value from CLIMAParameters because Ollie said it was too difficult to change the default to be a String. (See ZenithAngleCalc.jl where I have `epoch_string`.) I’m not sure how to do it exactly, but you’ll have to update the `create_parameters.jl` file to not just take the defaults from CLIMAParameters. Honestly for the other orbital parameters it might just be worth having those local to this package as well since nothing else in the code should need to know the eccentricity of the orbit (for example). But if this is tricky for some reason, we can change it in a future PR when I’m back!

File: src/ZenithAngleCalc.jl [hunk 11]
```diff
@@ -17,18 +17,18 @@ end
...
-    day_length::FT = Planet.day(param_set)
-    d0::FT = orbit_semimaj(param_set)
-    M0::FT = mean_anom_epoch(param_set)
-    
+    Ya::FT = IP.year_anom(param_set)
+    day_length::FT = IP.day(param_set)
+    d0::FT = IP.orbit_semimaj(param_set)
+    M0::FT = IP.mean_anom_epoch(param_set)
+
     epoch_string = "2000-01-01T11:58:56.816"
```
Comment (claresinger) [inline src/ZenithAngleCalc.jl:28]: This is where `epoch_string` is currently defined which I know is not ideal.

**Review by claresinger — APPROVED** [review]
Thank you @charleskawczynski. Looks good. Just some small notes about types and `IP.APS` vs. `AIP`. And then one bigger change which is adding in the String definition of `epoch` to this local parameters creation. But otherwise tests are passing and docs look fine so it should be okay for upstream use by ClimaAtmos as is.

File: docs/src/InsolationExamples.md (hunk 7 above)
Comment (charleskawczynski) [reply docs/src/InsolationExamples.md:27]: It is. Well, this line is actually repeated in `include(joinpath(pkgdir(Insolation), "parameters", "create_parameters.jl"))` (because it's still needed in order to create `param_set`), so we can remove this one. I'm changing `using` to `import` because it's better practice.

Comment (charleskawczynski) [reply docs/src/InsolationExamples.md:27]: I'll delete this duplicate line.

File: docs/src/InsolationExamples.md (hunk 1 above)
Comment (charleskawczynski) [reply docs/src/InsolationExamples.md:31]: It should work fine with Float32. This example was previously using Float64, so I kept it unchanged. I'm just declaring Float64 here so that it's transparent (`create_insolation_parameters` requires the float type in order to make the new parameter structs).

File: docs/src/InsolationExamples.md (hunk 2 above)
Comment (charleskawczynski) [reply docs/src/InsolationExamples.md:53]: Same as above

File: docs/src/InsolationExamples.md (hunk 3 above)
Comment (charleskawczynski) [reply docs/src/InsolationExamples.md:64]: Yeah, I think I just wanted to make sure the plots looked okay.

File: docs/src/InsolationExamples.md (hunk 4 above)
Comment (charleskawczynski) [reply docs/src/InsolationExamples.md:83]: Same as above

File: docs/src/find_equinox_perihelion_dates.jl (hunk 5 above)
Comment (charleskawczynski) [reply docs/src/find_equinox_perihelion_dates.jl:7]: Same as above

File: docs/src/plot_diurnal_cycle.jl (hunk 6 above)
Comment (charleskawczynski) [reply docs/src/plot_diurnal_cycle.jl:9]: Same as above

File: src/Insolation.jl (hunk 8 above)
Comment (charleskawczynski) [reply src/Insolation.jl:7]: Yeah, I was a bit inconsistent with this, we can change `APS` to `AIP` (and `IP.AIP`).

File: src/Parameters.jl (hunk 9 above)
Comment (charleskawczynski) [reply src/Parameters.jl:7]: Yes, absolutely, done!

File: src/Parameters.jl (hunk 10 above)
Comment (charleskawczynski) [reply src/Parameters.jl:15]: Right, the string bit is a bit difficult at the moment. Let's tackle this in a later PR, since this one is already pretty busy.

File: src/ZenithAngleCalc.jl (hunk 11 above)
Comment (charleskawczynski) [reply src/ZenithAngleCalc.jl:28]: Let's tackle this in a later PR, since this one is already pretty busy.

**Comment by charleskawczynski** [comment]
Thanks for the review @claresinger!

**Comment by charleskawczynski** [comment]
I forgot to update the bors toml
